x-pack/plugins/actions/server/builtin_action_types/es_index.test.ts

```diff
@@ -305,12 +312,240 @@ describe('execute()', () => {
+        ],
+        "indexOverride": null,
+      }
+    `);
+  });
+
+  test('ignores indexOverride for generic es index connector', async () => {
+    expect(actionType.renderParameterTemplates).toBeTruthy();
+    const paramsWithTemplates = {
+      documents: [{ hello: '{{who}}' }],
+      indexOverride: 'hello-world',
+    };
+    const variables = {
+      who: 'world',
+    };
+    const renderedParams = actionType.renderParameterTemplates!(
+      paramsWithTemplates,
+      variables,
+      'action-type-id'
+    );
+    expect(renderedParams).toMatchInlineSnapshot(`
+      Object {
+        "documents": Array [
+          Object {
+            "hello": "world",
+          },
+        ],
+        "indexOverride": null,
+      }
+    `);
+  });
+
+  test('renders parameter templates as expected for preconfigured alert history connector', async () => {
+    expect(actionType.renderParameterTemplates).toBeTruthy();
+    const paramsWithTemplates = {
+      documents: [{ hello: '{{who}}' }],
+      indexOverride: null,
+    };
+    const variables = {
+      date: '2021-01-01T00:00:00.000Z',
+      rule: {
+        id: 'rule-id',
+        name: 'rule-name',
+        type: 'rule-type',
+      },
+      context: {
+        contextVar1: 'contextValue1',
+        contextVar2: 'contextValue2',
+      },
+      params: {
+        ruleParam: 1,
+        ruleParamString: 'another param',
+      },
+      tags: ['abc', 'xyz'],
+      alert: {
+        id: 'alert-id',
+        actionGroup: 'action-group-id',
+        actionGroupName: 'Action Group',
+      },
+      state: {
+        alertStateValue: true,
+        alertStateAnotherValue: 'yes',
+      },
+    };
+    const renderedParams = actionType.renderParameterTemplates!(
+      paramsWithTemplates,
+      variables,
+      AlertHistoryEsIndexConnectorId
+    );
+    expect(renderedParams).toMatchInlineSnapshot(`
+      Object {
+        "documents": Array [
+          Object {
+            "@timestamp": "2021-01-01T00:00:00.000Z",
+            "alert": Object {
+              "actionGroup": "action-group-id",
+              "actionGroupName": "Action Group",
+              "context": Object {
+                "rule-type": Object {
+                  "contextVar1": "contextValue1",
+                  "contextVar2": "contextValue2",
+                },
+              },
+              "id": "alert-id",
+            },
+            "event": Object {
+              "kind": "alert",
+            },
+            "rule": Object {
+              "id": "rule-id",
+              "name": "rule-name",
+              "params": Object {
+                "rule-type": Object {
+                  "ruleParam": 1,
+                  "ruleParamString": "another param",
+                },
+              },
+              "type": "rule-type",
+            },
+            "tags": Array [
+              "abc",
+              "xyz",
+            ],
+          },
+        ],
+        "indexOverride": null,
```

Comment:
I used `schema.nullable` instead of `schema.maybe` for `indexOverride` because I thought we were trying to get away from `schema.maybe` so am required to return a `string | null` for the `indexOverride` value

Reply:
Ya, but that's just for our usage within alerting; what we actually WRITE to these new indices _could_ be different.  Just wondering, in the name of cleanliness, whether we should remove this in those indices, since it's basically a hidden property. Presumably it's the same value as the  `_index` value available in the document.

I'm thinking - let's scratch that thought - perhaps someday there'd be some value to having it there, and then we'd have to add it back.  And it's more code, and could be confusing to one of us why we removed it anyway.

We could look at removing it later ...